src/rust/plugin-bootstrap/src/plugin_bootstrap_init.rs

```diff
@@ -0,0 +1,41 @@
+use std::os::unix::fs::PermissionsExt;
+
+use plugin_bootstrap::client::PluginBootstrapClient;
+use rust_proto::plugin_bootstrap::{
+    GetBootstrapInfoRequest,
+    GetBootstrapInfoResponse,
+};
+
+static PLUGIN_BINARY_PATH: &str = "/usr/local/bin/grapl-plugin";
+static CLIENT_CERTIFICATE_PATH: &str = "/etc/ssl/private/plugin-client-cert.pem";
+static PLUGIN_CONFIG_PATH: &str = "/etc/systemd/system/plugin.service.d/override.conf";
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let (_env, _guard) = grapl_config::init_grapl_env!();
+
+    let mut bootstrap_client = PluginBootstrapClient::from_env().await?;
+    let GetBootstrapInfoResponse {
+        plugin_payload,
+        client_certificate,
+    } = bootstrap_client
+        .get_bootstrap_info(GetBootstrapInfoRequest {})
+        .await?;
+
+    std::fs::write(PLUGIN_BINARY_PATH, plugin_payload.plugin_binary)?;
+    std::fs::write(
+        CLIENT_CERTIFICATE_PATH,
+        client_certificate.client_certificate,
+    )?;
+
+    std::fs::set_permissions(PLUGIN_BINARY_PATH, std::fs::Permissions::from_mode(0o755))?;
```

Comment:
Wait.. this file was already written on L25.. Do we ever need to write it again? If not then either `0o100` or `0o010` should suffice, I think?

Reply:
Just noting the user executing the above code is currently root [1], and the plugin is to be executed by a dynamic user [2], provided by systemd.

[1] https://github.com/grapl-security/grapl/pull/1413/files#diff-0781f24f5e0c738f4ac02482ba5498c3804068df157b21b4a6945a8460d99b31R9-R10
[2] https://github.com/grapl-security/grapl/pull/1413/files#diff-d35b307570920bc59095340f673266b7f73e3b38bac0d821223702499ddec833R26

Reply:
Hmm so the user doesn't belong to any particular group, in which case the binary needs to be `0o001` (executable by "other", readable by none, writable by none).. correct?

Reply:
Another thing that came up in slack is that this "binary" may not be.. a binary (e.g. a compiled computer program). Instead it could be a `.pex` file. I think maybe the word "binary" here could be a little confusing in that case.

Reply:
FWIW the permissions here aren't much of a security control since our isolation boundaries assume the plugin is the malicious actor - the permissions were more to let me execute binaries and that sort of thing.

So we shouldn't worry too much about it and instead just go with something minimal but working.